### PR TITLE
DPDK: Check for correct cabling

### LINKED ISSUE #487: Incorrectly cabled nodes fail to exit
There appears to be a serious problem where it is possible for a node to be cabled incorrectly in a way that won't cause kotekan to shutdown and warn about it.   To make matters worse this bug also leads the `StreamID` to get set to `[0,0,0,0]` or `bin = 0` for all the data it is generating, possibly overwriting or duplicating existing good data for `bin = 0`.

### PR DESCRIPTION
**Pull request by andrerenard**
Checks if the cabling (streamIDs) meets the expected pattern for a correct shuffle.  This should resolve the bug in the last run where the system created duplicate frequency products. 

Closes #487

### CODE REVIEW

**Comment by andrerenard** [comment]
Note this needs a little more testing before I merge it in.

File: lib/dpdk/iceBoardShuffle.hpp [hunk 1]
```diff
@@ -308,6 +326,46 @@ inline int iceBoardShuffle::handle_packet(struct rte_mbuf* mbuf) {
...
+        if (i == port || all_stream_ids[i].crate_id == 255)
+            continue;
+
+        // Check that all the slots and links are the same.
+        if (all_stream_ids[i].slot_id != slot_id || all_stream_ids[i].link_id != link_id) {
+            FATAL_ERROR("One of the link_ids or slot_ids don't match! There is a cabling problem.");
+            return false;
+        }
+
+        // Check that we don't have the same create ID as another link
```
Comment (james-s-willis) [inline lib/dpdk/iceBoardShuffle.hpp]: Typo.

File: lib/dpdk/iceBoardShuffle.hpp [hunk 2]
```diff
@@ -308,6 +326,46 @@ inline int iceBoardShuffle::handle_packet(struct rte_mbuf* mbuf) {
...
+        // This should be impossible unless there is an FPGA problem
+        if (all_stream_ids[i].crate_id == crate_id) {
+            FATAL_ERROR("Two of the crate_ids are the same! There is a cabling problem.");
+            return false;
+        }
+
+        // Check that all the creates are from the same group (all even/odd)
```
Comment (james-s-willis) [inline lib/dpdk/iceBoardShuffle.hpp]: Typo.

File: lib/dpdk/iceBoardShuffle.hpp [hunk 3]
```diff
@@ -308,6 +326,46 @@ inline int iceBoardShuffle::handle_packet(struct rte_mbuf* mbuf) {
...
+        if (even != ((all_stream_ids[i].crate_id % 2) == 0)) {
+            FATAL_ERROR("The create IDs are not all even or all odd. There is a cabling problem.");
```
Comment (james-s-willis) [inline lib/dpdk/iceBoardShuffle.hpp]:
```suggestion
            FATAL_ERROR("The crate IDs are not all even or all odd. There is a cabling problem.");
```

**Pull request merged**